scripts/src/errors.ts

```diff
@@ -0,0 +1,129 @@
+export type ApiError = {
+	code: number;
+	error: string;
+	message: string;
+};
+
+export const CODES = {
+	Unauthorized: {
+		MissingToken: 1,
+		InvalidToken: 2
+	},
+	NotFound: {
+		App: 1,
+		Offer: 2,
+		Order: 3,
+		PublicKey: 4
+	},
+	Conflict: {
+		ExternalOrderExhausted: 1
+	},
+	Locked: {
+		OfferCapReached: 1
+	},
+	UnprocessableEntity: {
+		OpenedOrdersOnly: 1,
+		OpenedOrdersUnreturnable: 2
+	},
+	BadRequest: {
+		UnknownSignInType: 1,
+		WrongJWTAlgorithm: 2
+	}
+};
```

Comment:
Is these a way to make all these enums? Maybe even encode the final value to each type. i.e.: `NotFound.App: 4041`

Reply:
you can't have hierarchy with enums, and i started with encoding the final values, but then switched to the current version as i thought that the generic factory function should be responsible for the prefix code

Reply:
explain what?

Reply:
Yeah I saw that. So can you not export this class so no one can misuse this?